### PR TITLE
[runtime] improve runtime of tests

### LINKED ISSUE #6872: runtime of tests increased
**Description**

The runtime of tests increased after the change - #6751

The runtime increase for many tests, for example _test_buffer.py_:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/antonh/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/antonh/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

Test name | Runtime before a change | Runtime after a change
-- | -- | --
test_buffer_deployment | 12m 37s | 15m 17s
test_change_speed_cable[10000-1500-15m] | 1m 45s | 2m 20s
test_change_speed_cable[10000-1500-40m] | 1m 50s | 2m 20s
test_change_speed_cable[50000-1500-15m] | 1m 55s | 1m 58s
test_change_speed_cable[50000-1500-40m] | 1m 57s | 2m 25s

</body>

</html>

**Steps to reproduce the issue:**
1. Run test (test_buffer.py for example) before and after the change 6751 

**Describe the results you received:**
degradation in runtime

### PR DESCRIPTION
**Pull request by AntonHryshchuk**
Signed-off-by: Anton <antonh@nvidia.com>

### Description of PR

Summary:
Improve runtime of tests. The update of the asic service related to multi-devices only. 
Fixes #6872

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
short runtime of tests

#### How did you do it?
added "if" condition before removing service from DEFAULT_ASIC_SERVICES

#### How did you verify/test it?
executed and compared the runtime before and after the change

### CODE REVIEW

**Comment by AntonHryshchuk** [comment]
@abdosi please review

**Comment by liat-grozovik** [comment]
@abdosi LGTM but appreciate to get your feedback as it seems related to https://github.com/sonic-net/sonic-mgmt/pull/6751 
To me the fix seems crucial as it causing a degradation in single ASIC switch. Please prioritize the review.

File: tests/common/devices/multi_asic.py [hunk 1]
```diff
@@ -69,13 +69,14 @@ def critical_services_tracking_list(self):
...
                 active_asics.append(a_asic_instance)
         service_list += self._DEFAULT_SERVICES
-        # Update the asic service based on feature table state and asic flag
-        config_facts = self.config_facts(host=self.hostname, source="running")['ansible_facts']
-        for service in list(self.sonichost.DEFAULT_ASIC_SERVICES):
-            if config_facts['FEATURE'][service]['has_per_asic_scope'] == "False":
-                self.sonichost.DEFAULT_ASIC_SERVICES.remove(service)
-            if config_facts['FEATURE'][service]['state'] == "disabled":
-                self.sonichost.DEFAULT_ASIC_SERVICES.remove(service)
+        if self.sonichost.is_multi_asic:
```
Comment (abdosi) [inline tests/common/devices/multi_asic.py:72]: can we make it `is_supervisor_node()` that should be better.

**Review by abdosi — CHANGES_REQUESTED** [review]
as noted.

**Comment by liat-grozovik** [comment]
@abdosi  could you please approve following the fix provided?

**Comment by liat-grozovik** [comment]
> as noted.

comment was addressed. moving a head with merge

**Review by roysr-nv — APPROVED** [review]

**Review by abdosi — APPROVED** [review]

**Pull request merged**